Automattic/pocket-casts-android
# Merging UserSettings to main

## Description
This is merging to `main` the refactoring of the to-be-synced settings to be UserSettings.

## Testing Instructions
All of these changes have already been tested in separate PRs, so they shouldn't need to be retested individually.

The one exception to this is the settings on the Notification screen because I had to resolve some non-trivial merge conflicts when merging in main due to https://github.com/Automattic/pocket-casts-android/pull/1264 and https://github.com/Automattic/pocket-casts-android/pull/1244. It's worth giving those another quick test to just make sure there aren't any issues.

> **Warning**
> 7.46 has been cut, so this is no longer relevant. ~This can be reviewed now, but let's not merge it until after the 7.46 release is cut since the number of changes in here creates a bit of a risk that there could be a regression, so it would be good to have as much time as possible to catch any issues.~

## Checklist
